### ISSUE TITLE
Some default gRPC request examples do not have default fields set.

### ISSUE THREAD

**Issue by trustin** [line/armeria#2690]
For example, `CreateGreetingRequest` is serialized into an empty JSON object when `GrpcDocServicePlugin` builds its default example request:

```protobuf
message CreateGreetingRequest {
    FullName full_name = 1;
}
message FullName {
    string first_name = 1;
    string last_name = 2;
}
```

A workaround is to specify an example object explicitly when building a `DocService`, as documented in https://line.github.io/armeria/server-docservice.html#example-requests-and-headers

/cc @gary-lo

**Comment by anuraaga** [comment]
Hmm - default value for a message field is to not be present so this is technically correct as far as protobuf's json spec goes. If we want to show deeply, we'll need to fill a proto with empty submessages using proto reflection.

**Comment by trustin** [comment]
I think we should generate a deep one in this case of providing a skeletal default example. Could be done probably by implementing our own printer or discussing about adding a new option to the upstream.

**Comment by anuraaga** [comment]
Yeah makes sense. I don't think we need a custom printer, just need to populate the protocol with empty submessages which should be less code.

**Comment by trustin** [comment]
> just need to populate the protobuf with empty submessages which should be less code.

Is it doable with `DynamicMessage`? That'd be nice if so.

**Comment by gary-lo** [comment]
I wasn't aware that of this functionality -
https://line.github.io/armeria/server-docservice.html#example-requests-and-headers

Possibly making this more obvious when the empty sample exists. on the front-end might be all that's required?
![image](https://user-images.githubusercontent.com/12748701/81245326-07c38f80-9058-11ea-978c-4e93c9576e10.png)

**Comment by anuraaga** [comment]
@trustin Don't think we even need `DynamicMessage`, just use the descriptor to find fields that are message type. But after thinking a bit more, I'm still hesitant to deep expand the fields. Logic can be quite different based on the presence of a submessage or not (not the case for primitive fields), so hitting Submit with the default can produce fairly wild results. And dealing with one-ofs will be even trickier - ignoring oneof, we'd fill every submessage field and have an unparsable proto. Due to this complexity and the fact that the only well defined behavior is following the proto semantics correctly, anything else is a questionable heuristic, I suggest just leaving as is.

@gary-lo Yeah, especially for complex protos, I think defining the examples yourself will be more helpful to your debuggers anyways. So by not obvious, you mean that an empty sample being black and a non-empty one being grey isn't obvious? Yeah it does seem tricky, any suggestions? 😆 Actually, I think implementing #43 could help a lot too, right now it's pretty annoying understanding the structure of the proto to fill it in correctly, example or not.

**Comment by trustin** [comment]
@anuraaga wrote:
> I suggest just leaving as is.
Agreed.

@gary-lo wrote:
> Possibly making this more obvious when the empty sample exists. on the front-end might be all that's required?
That's a good idea. Let me leave this issue open until 1) our new web site lands and 2) the frontend is updated.

**Comment by ikhoon** [comment]
Related work: https://github.com/fullstorydev/grpcui#request-form handles one-of nicely.
![image](https://user-images.githubusercontent.com/1866157/85253224-aab15b00-b498-11ea-9669-4f3dd3684d66.png)

**Comment by trustin** [comment]
How about creating a new issue for revamping the debug page? I think the original issue has been resolved via #2828.

**Comment by ikhoon** [comment]
> How about creating a new issue for revamping the debug page? I think the original issue has been resolved via #2828.

Yeap, let me close this issue and create anthoer one for the debug page.
